### PR TITLE
feat(data)!: the transform release line — one Resample, Python workflows, transform backend

### PR DESCRIPTION
**Pull request by vboussot**
**The whole transform-release line, as one linear PR.** The stack (#91, #93–#97, #99) was reviewed slice by slice — CI green, every CodeRabbit thread addressed — then a top-down merge cascaded the slices into each other instead of into main and GitHub closed the intermediates. This PR now carries the identical, byte-verified 36-commit line against main; the leftover `feat/impact-reg-cases-and-fields` and `feat/itktransform-backend` branches hold the same tree plus cascade merge commits and can be deleted after this merges.

One `Resample` (stored transforms decoded and streamed, windows measured at run, no bound to declare), the workflows as Python callables, the `:itktransform` Dataset backend + streamed NIfTI writes, the impact-reg orchestrator fully out-of-core with presets that declare only their transform, and the review cycle's fixes across API, sampling, planning and docs.

Merge with "Rebase and merge" (or a merge commit if you prefer a single landmark) — the history is already linear.

### CODE REVIEW

File: konfai/data/transform.py [hunk 1]
```diff
@@ -897,617 +972,107 @@ def inverse(self, name: str, tensor: torch.Tensor, cache_attribute: dict[str, An
...
-class ResampleToShape(Resample):
-    def __init__(self, shape: list[float] = [100, 256, 256], inverse: bool = True) -> None:
-        super().__init__(inverse)
-        self.shape = torch.tensor([0 if s < 0 else s for s in shape])
+    def __init__(self, spacing: list[float] | None, shape: list[int] | None, align: str) -> None:
+        self.spacing = None if spacing is None else np.asarray([max(0.0, float(value)) for value in spacing])
+        self.shape = None if shape is None else tuple(max(0, int(value)) for value in shape)
+        self.align = align
+        # A density is meaningless without the density it starts from; a count is not.
+        self.needs = frozenset({"Spacing"}) if spacing is not None else frozenset()
```
[the remark this answers is not in the document]
Comment (vboussot) [reply konfai/data/transform.py:1031]: A value <= 0 is the keep-this-axis sentinel: `shape: [0, 16, 16]` rescales only the axes it names (pinned by `test_an_axis_the_map_leaves_alone_is_left_alone` and `test_resample_to_shape_does_not_mutate_config`). The clamp now says so in place rather than refusing a documented spelling.
